src/plugins/unified_search/public/actions/apply_filter_action.ts

```diff
@@ -22,7 +21,6 @@ export const ACTION_GLOBAL_APPLY_FILTER = 'ACTION_GLOBAL_APPLY_FILTER';
 export interface ApplyGlobalFilterActionContext {
   filters: Filter[];
   timeFieldName?: string;
-  embeddable?: IEmbeddable;
```

Comment:
@elastic/kibana-visualizations Is this okay to remove here? It resolves a circular dependency between unified search and embeddable plugins since I'm importing the `FilterItems` component from unified search.

Reply:
It looks like this prop has a history.

- [First](https://github.com/lukeelmers/kibana/commit/1ac56d7bfcd1b9df542422afbcf4b3e2caaafac3#diff-61caee3ee61740ef514e66706233ac53799226a458a8049c564ef524765975ac), it was typed
- [Then](https://github.com/elastic/kibana/pull/82791/files#diff-61caee3ee61740ef514e66706233ac53799226a458a8049c564ef524765975acR32-R34) it was made `unknown` to avoid a circular dependency
- [Then](https://github.com/elastic/kibana/issues/116894), Stacey objected
- [Then](https://github.com/elastic/kibana/pull/137808), it was typed again

If you're in a hurry, I would suggest we go back to `unknown` so that theoretical consumers at least know that there's a prop there.

But, I think the best answer here would be to create an `Embeddable` types package as [suggested here](https://github.com/elastic/kibana/issues/116894#issuecomment-1200040707).

WDYT?

Reply:
I think you're right about creating an `EmbeddableTypes` package being the best route here but should be tackled as a separate PR. For now, I will restore the type back to `embeddable?: unknown` just to unblock this PR.

@ThomThomson what do you think?

Reply:
Here's the issue https://github.com/elastic/kibana/issues/163994

Reply:
I undid this change, and the one in `x-pack/plugins/lens/public/trigger_actions/open_in_discover_drilldown.tsx` and I'm not seeing any circular dependency issues. I _think_ this works alright because we're only importing types in `apply_filter_action`.

Reply:
When I revert those two lines, I'm seeing this error in CI [build failure](https://buildkite.com/elastic/kibana-pull-request/builds/151045#0189fee2-94cf-42aa-ba1f-a40e137283db). 
<img width="877" alt="Screenshot 2023-08-16 at 8 24 14 AM" src="https://github.com/elastic/kibana/assets/1697105/b2f67a30-3771-42ab-adcb-f8e8898eb1d9">

Reply:
Looks like I was wrong here - it seems like using type only imports helps resolve the plugin circular dependencies, but we can still suffer from them in typescript when adding circular deps to a `tsconfig`. Let's use unknown for the time being, and keep https://github.com/elastic/kibana/issues/163994 open to track a resolution. Likely we can incorporate this type of fix into the new Embeddables system.